### ISSUE TITLE
MIPS 32-bit target without Atomic{I,U}64

### ISSUE THREAD

**Issue by Borber** [boa-dev/boa#2599]
https://github.com/Borber/seam/actions/runs/4154490037/jobs/7186917428#step:5:227 

How should I fix this problem

**Comment by MisterEggnog** [comment]
I looked into a bit, seems that the issue has to do with the fact that [64bit atomics are not available on mips platforms with 32bit pointers].

Unfortunately this is an issue with [rust-gc] so you may want to publish an issue there, however, while the repository is in continuous development they have not released an update in two years.

It should not be too difficult to fix, looking at the code they just need to add a few cfg guards for the 64bit atomics.

[64bit atomics are not available on mips platforms with 32bit pointers]: https://doc.rust-lang.org/std/sync/atomic/index.html
[rust-gc]: https://github.com/Manishearth/rust-gc

**Comment by jedel1043** [comment]
We actually do use 64 bit atomics for our `JsSymbol` implementation.

https://github.com/boa-dev/boa/blob/ffa854ce3f8c6dcfdb24721513e55a2a9245910d/boa_engine/src/symbol.rs#L49-L52

We'll probably have to change that to an `AtomicUSize`, which should be available on more platforms.

Also, we've already pulled `rust-gc` into `boa_gc` because we needed to add support for weak pointers, so any issue related to the GC must be filed on this project.
